src/site/_filters/algolia-item.js

```diff
@@ -19,13 +19,19 @@ const md5 = require('./md5');
 const {getDefaultUrl} = require('./urls');
 const {defaultLocale} = require('../_data/site');
 const {generateImgixSrc} = require('../_includes/components/Img');
+const {findByUrl} = require('./find-by-url');
 
 /**
  * @param {EleventyCollectionItem} post
  * @return {AlgoliaItem}
  */
 function algoliaItem(post) {
   const data = post.data;
+  const parentUrl = `${post.data.page.filePathStem
+    .split('/')
+    .slice(0, -2)
+    .join('/')}/`;
+
```

Comment:
As explained in the other comment, extract to it's own method, add a proper JSDoc comment and maybe add a test case for it, for known paths.